src/test/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInputTests.kt

```diff
@@ -91,7 +91,7 @@ class ClusterMetricsInputTests {
         // GIVEN
         path = "/_cluster/health/"
         pathParams = "index1,index2,index3,index4,index5"
-        url = "http://localhost:9200/_cluster/health/index1,index2,index3,index4,index5"
+        url = "http://localhost:9200/_cluster/health/index1%2Cindex2%2Cindex3%2Cindex4%2Cindex5"
```

Comment:
This feels unexpected. Did the API use to take URL-decoded URL and pass it as is, but now requires the arguments to be url-encoded? Either way I think you want to be explicit and call a URL encode function instead of changing all `,` by `%2C`.

Reply:
@dblock fixed the logic so tests no longer need changes. https://github.com/opensearch-project/common-utils/blob/5a0b10dd4af42abd4099e846328deaeefc2166f5/src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt#L208-#L218
Looks like it is a problem with `httpcomponents 5.1.x` library as it has inconsistent behavior across 2 of its constructors.

this encodes the path params
```
val uriBuilder = URIBuilder()
                .setScheme(SUPPORTED_SCHEME)
                .setHost(SUPPORTED_HOST)
                .setPort(SUPPORTED_PORT)
                .setPath(path + pathParams)
```

while this doesnt

```
val uriBuilder = URIBuilder("$SUPPORTED_SCHEME://$SUPPORTED_HOST:$SUPPORTED_PORT$path$pathParams")
```
As this class is used by `alerting` plugin i would like to keep the tests same as any change in behavior may break `alerting` cluster metrics scenario.
Kindly have a look.